### ISSUE TITLE
General Drakkisath loot table

### ISSUE THREAD

**Issue by Klala** [dalaranwow/dalaran-wow#4681]
**Current behaviour**: Currently General Drakkisath loot table is missing two items: 
Pattern: Red Dragonscale Breastplate and Blackblade of Shahram I've been farming him for 100+ kill and they never dropped, While Pattern: Red Dragonscale Breastplate might be binds when picked up and only a leatherworker can see it drop, it is still a recipe from vanilla and I don't remember that looting systen in vanilla had that mechanic, other bop items and recipes drop from mobs / chests for everyone and this shouldn't be any different. The major issue though is the Blackblade of Shahram which I've checked a couple of wotlk/tbc/vanilla databases and none have shown less than 1% on that item, most show 2% or 4%.

**Expected behaviour**: Both the Pattern: Red Dragonscale Breastplate and Blackblade of Shahram should drop from General Drakkisath.

**Steps to reproduce the problem**:

Check the database for the fastest check I'd say or:
1. Go to UBRS
2. Kill General Drakkisath.
3. Notice that those items don't drop

**Comment by Edzero** [comment]
Blackblade of Shahram is meant to have a 1% drop rate in this patch.

WoW databses such as OpenWoW do not always show the correct, intended drop rates. They show what the default drop rates are in the core upon which the database is built.

**Comment by Klala** [comment]
I wasn't looking in openwow because that database is simply wrong, no server should follow the details there.
Indeed 1% is what it should be, but again, I've killed him 100+ times and still going and no sign of that sword or the recipe. Something is off and the best way to check that is reporting this here and get answers and/or fixes. I wish I could have anyone that has the server's database within their view just so I could ask them and avoid making an issue here.

While I do understand and know that any internet database can't show me the intended % of any item, I am forced to check the internet anyway because there is just no way for me to view Dalaran's database on my own.

**Comment by ashido96** [comment]
If I remember correctly, doing 1% chance 100 times gives approximately a 63% chance of receiving the item. Not getting it is reasonable.

**Comment by Klala** [comment]
I guess but I'm gonna keep trying. Up until now even 0.5% was less than 100 kills before it dropped. Right now I'm nearing the 120 kill so I will update here.

**Comment by TeridanDW** [comment]
Hello, thank you for the report @Klala , however in future reports please add the **ids** of items/creatures/gameobjects please.

Regarding the issue:
 - item http://www.wowhead.com/item=15730/pattern-red-dragonscale-breastplate has 4% drop chance (in my DB) and i could only assume it's the same for dalaran's realms.
Screenshot from PTR realm, I got it at 5-th or 6-th kill:
![image](https://cloud.githubusercontent.com/assets/6797611/25083952/b229329c-2362-11e7-88fb-399dbd91bb54.png)

 - item http://www.wowhead.com/item=12592/blackblade-of-shahram is missing from the loot table of the General (in my DB) and i could only assume it's the same for dalaran's realms.


> Blackblade of Shahram is meant to have a 1% drop rate in this patch.

All of the information that i could find matches this value above.

I did some tests on the sword (been added with GM command) and the proc does seem to work properly however the creature that is summoned is broken (doing 30-40 damage autoattacks, no spells, aggressive state).
![image](https://cloud.githubusercontent.com/assets/6797611/25084224/15977b30-2364-11e7-81b8-c2e0624cb3cd.png)

I will try to script this creature however if someone could find **_ANY_** information for him and how he is suppose to behaive it would be of great help. Thank you!
The creature is http://www.wowhead.com/npc=10718/shahram
And is been summoned by spell http://www.wowhead.com/spell=16602/shahram

**Comment by ashido96** [comment]
http://www.wowhead.com/item=12592/blackblade-of-shahram#comments

Top comment has some of the stuff he casts. No timers or logic on any of them of course.

**Comment by TeridanDW** [comment]
Thank you @ashido96 i saw that but my question was directed towards:

> timers or logic

however i think i found the answer here - http://wowwiki.wikia.com/wiki/Shahram

> When Blackblade of Shahram procs, Shahram is summoned. He casts a random spell and then disappears. The spells he can cast are:

**Comment by TeridanDW** [comment]
Notes.
Spell:
 - http://www.wowhead.com/spell=16598/will-of-shahram - healing part works only if the caster(shahram) is present, if he despawns the spell nolonger heals the targets(the effect is still present on the targets), however the mana regeneration works always.

 - http://www.wowhead.com/spell=16598/will-of-shahram - works as intended

 - http://www.wowhead.com/spell=16597/curse-of-shahram - works as intended

 - http://www.wowhead.com/spell=16600/might-of-shahram - works as intended

 - http://www.wowhead.com/spell=16601/fist-of-shahram - works as intended (note - spell gives 300 haste rating and that's diffrent _**%**_ based on the level of the player)

 - http://www.wowhead.com/spell=16596/flames-of-shahram - works as intended

**Comment by TeridanDW** [comment]
What is done here:
 - added item http://www.wowhead.com/item=12592/blackblade-of-shahram to the loot table of creature http://www.wowhead.com/npc=10363/general-drakkisath with _**1 %**_ drop chance
 - scripted creature http://www.wowhead.com/npc=10718/shahram which is summoned by spell http://www.wowhead.com/spell=16602/shahram which is casted by item http://www.wowhead.com/item=12592/blackblade-of-shahram

To do:
 - fix issue with spell http://www.wowhead.com/spell=16599/blessing-of-shahram : 

> healing part works only if the caster(shahram) is present, if he despawns the spell nolonger heals the targets(the effect is still present on the targets), however the mana regeneration works always.

How it looks now:
![image](https://cloud.githubusercontent.com/assets/6797611/25088304/0ea437c2-237e-11e7-8d5c-dc740b22253e.png)
![image](https://cloud.githubusercontent.com/assets/6797611/25088317/242a9e1a-237e-11e7-94b3-3a082fedfe4f.png)

``` sql
DELETE FROM `creature_loot_template` WHERE `entry` = 10363 AND `item`=12592;
INSERT INTO `creature_loot_template` (`Entry`, `Item`, `Reference`, `Chance`, `QuestRequired`, `LootMode`, `GroupId`, `MinCount`, `MaxCount`, `Comment`) VALUES
(10363,12592,0,1,0,1,0,1,1,NULL);

UPDATE `creature_template` SET `unit_flags` = 33685504 WHERE `entry` = 10718; 

-- Shahram SAI
SET @ENTRY := 10718;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,0,0,1,54,0,100,0,0,0,0,0,8,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"Shahram - On Just Summoned - Set Reactstate Passive"),
(@ENTRY,0,1,2,61,0,100,0,0,0,0,0,103,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Shahram - On Just Summoned - Set Rooted On"),
(@ENTRY,0,2,0,61,0,100,0,0,0,0,0,87,@ENTRY*100+00,@ENTRY*100+01,@ENTRY*100+02,@ENTRY*100+03,@ENTRY*100+04,@ENTRY*100+05,1,0,0,0,0,0,0,0,"Shahram - On Just Summoned - Run Random Script");

-- Actionlist SAI
SET @ENTRY := 1071800;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,0,0,0,0,11,16599,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Cast 'Blessing of Shahram'"),
(@ENTRY,9,1,0,0,0,100,0,0,0,0,0,41,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Despawn Instant");

-- Actionlist SAI
SET @ENTRY := 1071801;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,0,0,0,0,11,16598,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Cast 'Will of Shahram'"),
(@ENTRY,9,1,0,0,0,100,0,0,0,0,0,41,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Despawn Instant");

-- Actionlist SAI
SET @ENTRY := 1071802;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,0,0,0,0,11,16597,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Cast 'Curse of Shahram'"),
(@ENTRY,9,1,0,0,0,100,0,0,0,0,0,41,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Despawn Instant");

-- Actionlist SAI
SET @ENTRY := 1071803;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,0,0,0,0,11,16600,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Cast 'Might of Shahram'"),
(@ENTRY,9,1,0,0,0,100,0,0,0,0,0,41,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Despawn Instant");

-- Actionlist SAI
SET @ENTRY := 1071804;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,0,0,0,0,11,16601,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Cast 'Fist of Shahram'"),
(@ENTRY,9,1,0,0,0,100,0,0,0,0,0,41,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Despawn Instant");

-- Actionlist SAI
SET @ENTRY := 1071805;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,0,0,0,0,11,16596,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Cast 'Flames of Shahram'"),
(@ENTRY,9,1,0,0,0,100,0,0,0,0,0,41,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Despawn Instant");

```

**Comment by Klala** [comment]
@TeridanDW If I understand correctly you're not using the server's DB, you're using some other DB to validate what your said? Would you be kind enough to share that DB or is it something that's not on the web?
From what I can tell from the picture you put, you're a leatherworker so if I did 120 runs now and never seen it once it's either really bad luck or that recipe has LW skill requirement for one to see it.
I am pretty sure that recipe should drop with no requirements.

**Comment by TeridanDW** [comment]
Few more tests on fresh character.
 - Regarding:

>  - item http://www.wowhead.com/item=12592/blackblade-of-shahram is missing from the loot table of the General (in my DB) and i could only assume it's the same for dalaran's realms.

This seems to be wrong as i got the item on PTR realm:
![image](https://cloud.githubusercontent.com/assets/6797611/25125986/5ccab4b0-2439-11e7-907c-095800ea5386.png)

So will remove:

>  - added item http://www.wowhead.com/item=12592/blackblade-of-shahram to the loot table of creature http://www.wowhead.com/npc=10363/general-drakkisath with _**1 %**_ drop chance

from the code above as it is not needed.

 - This:

> From what I can tell from the picture you put, you're a leatherworker so if I did 120 runs now and never seen it once it's either really bad luck or that recipe has LW skill requirement for one to see it.

is true, my character (on the picture above (comment 6)) had leatherworking. After 30 mins of killing general (on fresh character without leatherworking) i didn't get the recipe also. As soon as I learn the proffesion I got it after 20-30 attempts.

> I am pretty sure that recipe should drop with no requirements.

Why do you think that it's currently wrong (to have proffesion requirement)? Providing a proof of that statement will be great.

 - Regarding:

> @TeridanDW If I understand correctly you're not using the server's DB, you're using some other DB to validate what your said? Would you be kind enough to share that DB or is it something that's not on the web?

I am using Dalaran's PTR realm for testing purposes and I set up my own realm for development purposes. The DB is diffrent for them and i have access only to my own realm's DB. With that been said you can set up your own realm(as i do) and check the loot tables or whatever else you want.
The DB that i use is the public one here:
TDB 335.62 https://github.com/TrinityCore/TrinityCore/releases
however you will need so set up a realm (i think) in order you view it.

Setting up a realm is not that hard at all, a detailed installation guide can be found here:
https://trinitycore.atlassian.net/wiki/display/tc/Installation+Guide
follow the steps 1 by 1 and you will be done in few hours.

**Comment by Klala** [comment]
@TeridanDW 

> Why do you think that it's currently wrong (to have proffesion requirement)? Providing a proof of that statement will be great.

I have no evidence of this other than my knowledge and experience with vanilla recipes.
Pre-tbc profession recipes didn't have profession requirement mechanic system. That was silently introduced in tbc, you could run in every other dungeon/raid and get inappropriate items which nobody could use, you'd just sell them to a vendor or toss them.
A good example for this is in Dire Maul north, when you loot the Knot Thimblejack's Cache you can get LW recipes even when there are no LW in your group.

**Comment by Rushor** [comment]
DB/Creature: Shahram - Add several castevent

* closes: https://github.com/dalaranwow/dalaran-wow/issues/4681
* added the castevent for the summoned creature. Please open up a new issue for the missing spellscript/condition of http://www.wowhead.com/spell=16599/blessing-of-shahram 
* commit: 00500e8d182173d7bf5d96c62c176d3f9c0ae0e5

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

Concerning the loot chance: 

Item 12592 drops in the reference loot template 35025 of the creature 10363 for 2 percent each kill.

**Comment by Rushor** [comment]
already on PTR
